design/design-doc-custom-composition.md

```diff
@@ -0,0 +1,179 @@
+# Custom Composition Types
+
+* Owner: Muvaffak Onuş (@muvaf)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+## Proposal
```

Comment:
```suggestion
## Proposal
Enable cdk8s users to offer `Constructs` as native CRDs via Crossplane `CompositeResourceDefinitions` (XRDs) and `CustomComposition` resources that can use sidecar images generated by the new `cdk8s pack` command that bundles the cdk8s program as a simple web server that can be used from a Crossplane `Composition` resource to emit the yaml for a Crossplane `CustomComposition` (e.g. https://github.com/eladb/cdk8s-pack-prototype). The emitted yaml resources would be labelled so `kubectl apply --prune` semantics can be observed.
```

Reply:
Should we move this under a goals heading?